lib/ansible/modules/identity/keycloak/keycloak_clienttemplate.py

```diff
@@ -0,0 +1,386 @@
+        required: false
+
+    name:
+        description:
+            - Name of the client template
+        required: false
+
+    description:
+        description:
+            - Description of the client template in Keycloak
+        required: false
+
+    bearer_only:
+        description:
+            - The access type of this client template is bearer-only.
+              This is 'bearerOnly' in the Keycloak REST API.
+        required: false
+
+    consent_required:
+        description:
+            - If enabled, users have to consent to client access.
+              This is 'consentRequired' in the Keycloak REST API.
+        required: false
+
+    standard_flow_enabled:
+        description:
+            - Enable standard flow for this client template or not (OpenID connect).
+              This is 'standardFlowEnabled' in the Keycloak REST API.
+        required: false
+
+    implicit_flow_enabled:
+        description:
+            - Enable implicit flow for this client template or not (OpenID connect).
+              This is 'implicitFlowEnabled' in the Keycloak REST API.
+        required: false
+
+    direct_access_grants_enabled:
+        description:
+            - Are direct access grants enabled for this client template or not (OpenID connect).
+              This is 'directAccessGrantsEnabled' in the Keycloak REST API.
+        required: false
+
+    service_accounts_enabled:
+        description:
+            - Are service accounts enabled for this client template or not (OpenID connect).
+              This is 'serviceAccountsEnabled' in the Keycloak REST API.
+        required: false
+
+    public_client:
+        description:
+            - Is the access type for this client template public or not.
+              This is 'publicClient' in the Keycloak REST API.
+        required: false
+
+    frontchannel_logout:
+        description:
+            - Is frontchannel logout enabled for this client template or not.
+              This is 'frontchannelLogout' in the Keycloak REST API.
+        required: false
+
+    protocol:
+        description:
+            - Type of client template (either C(openid-connect) or C(saml).
+        required: false
+        choices: ['openid-connect', 'saml']
+
+    full_scope_allowed:
+        description:
+            - Is the "Full Scope Allowed" feature set for this client template or not.
+              This is 'fullScopeAllowed' in the Keycloak REST API.
+        required: false
+
+    protocol_mappers:
+        description:
+            - a list of dicts defining protocol mappers for this client template.
+              An example of one is given in the examples section.
+              This is 'protocolMappers' in the Keycloak REST API.
+        required: false
+
+    attributes:
+        description:
+            - A dict of further attributes for this client template. This can contain various
+              configuration settings.
+        required: false
```

Comment:
This argument falls under the same guideline requirements as explained for `protocol_mappers`, this needs to be a more explicit argument, and not an arbitrary field.

Reply:
This field, likewise, is not further documented in the official Keycloak API documentation (http://www.keycloak.org/docs-api/3.4/rest-api/index.html#_clienttemplaterepresentation). Here, likewise, I can do a bit of digging/documenting for common options, but I do not think an exhaustive list is easy to do (or indeed to guarantee spanning API versions). Is there leeway here?

Reply:
Looking into it a bit further, I have not found this field used at all in client templates, so far. It is being used for clients (which are handled by keycloak_client, to which I could add some validation later on).

Is it prudent to remove this field completely, or should it be kept in case somebody does have an (undocumented) use for it?